chiselFrontend/src/main/scala/chisel3/internal/Builder.scala

```diff
@@ -95,6 +106,27 @@ private[chisel3] trait HasId {
   private[chisel3] def setRef(parent: HasId, index: Int): Unit = setRef(Index(Node(parent), ILit(index)))
   private[chisel3] def setRef(parent: HasId, index: UInt): Unit = setRef(Index(Node(parent), index.ref))
   private[chisel3] def getRef: Arg = _ref.get
+
+  // Implementation of public methods.
+  def signalName = _parent match {
+    case None => getRef.name
```

Comment:
On Fri, Aug 19, 2016 at 12:55 PM, Richard Lin <notifications@github.com>
wrote:

> In chiselFrontend/src/main/scala/chisel3/internal/Builder.scala
> <https://github.com/ucb-bar/chisel3/pull/259#discussion_r75542313>:
>
> > @@ -95,6 +106,27 @@ private[chisel3] trait HasId {
> >    private[chisel3] def setRef(parent: HasId, index: Int): Unit = setRef(Index(Node(parent), ILit(index)))
> >    private[chisel3] def setRef(parent: HasId, index: UInt): Unit = setRef(Index(Node(parent), index.ref))
> >    private[chisel3] def getRef: Arg = _ref.get
> > +
> > +  // Implementation of public methods.
> > +  def signalName = _parent match {
> > +    case None => getRef.name
>
> Is there ever a case where a signal wouldn't have a parent (other than
> because of bugs)? Should it assert out rather than trying to make something
> up?
>
> Yes, It shoudn't happen, so now exceptions are thrown in this case.

> I also assume that this is trying to be consistent with how CHIRRTL names
> are generated. If so, it would be nice to just reference the CHIRRTL
> generated name, as opposed to possibly duplicating logic here (like
> concatenating path and signal name in pathName implementation).
>
 As far as I know, there are no counter part names in CHIRRTL/FIRRTL for
pathName.

> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ucb-bar/chisel3/pull/259/files/d794926b9e06e882012d145868e4f8d64d2f367f#r75542313>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFIz8CkRjDNB2_TY3QRoyAKrlzFEhgx6ks5qhgoggaJpZM4JlDoC>
> .
>